src/wallet/test/fuzz/coinselection.cpp

```diff
@@ -74,6 +74,13 @@ static SelectionResult ManualSelection(std::vector<COutput>& utxos, const CAmoun
     return result;
 }
 
+static CTxOut GetChangePrototypeTxout(FuzzedDataProvider& fuzzed_data_provider)
+{
+    const CTxDestination tx_destination{ConsumeTxDestination(fuzzed_data_provider)};
+    CScript script_change{GetScriptForDestination(tx_destination)};
+    return CTxOut(0, script_change);
+}
```

Comment:
This will narrow down the range of output sizes being fuzzed.

An alternative would perhaps be to randomly roll change output size and change input size, while calculating all the other derived values from those, but my gut says that there shouldn’t be all that much exciting happening anyway. Will revisit this question when I’ve fuzzed a bit.

Reply:
Did you revisit it?

Reply:
> This will narrow down the range of output sizes being fuzzed.

Agree.
You don't need to create valid known destinations, we only care about the change output size. All yours:
```diff
diff --git a/src/wallet/test/fuzz/coinselection.cpp b/src/wallet/test/fuzz/coinselection.cpp
--- a/src/wallet/test/fuzz/coinselection.cpp	(revision ccb25a1ddc1f79f432a6718604f01ed97dced15f)
+++ b/src/wallet/test/fuzz/coinselection.cpp	(date 1702581796902)
@@ -11,6 +11,7 @@
 #include <test/util/setup_common.h>
 #include <wallet/coinselection.h>
 
+#include <algorithm>
 #include <vector>
 
 namespace wallet {
@@ -74,13 +75,6 @@
     return result;
 }
 
-static CTxOut GetChangePrototypeTxout(FuzzedDataProvider& fuzzed_data_provider)
-{
-    const CTxDestination tx_destination{ConsumeTxDestination(fuzzed_data_provider)};
-    CScript script_change{GetScriptForDestination(tx_destination)};
-    return CTxOut(0, script_change);
-}
-
 // Returns true if the result contains an error and the message is not empty
 static bool HasErrorMsg(const util::Result<SelectionResult>& res) { return !util::ErrorString(res).empty(); }
 
@@ -101,14 +95,16 @@
     coin_params.m_subtract_fee_outputs = subtract_fee_outputs;
     coin_params.m_long_term_feerate = long_term_fee_rate;
     coin_params.m_effective_feerate = effective_fee_rate;
-    auto change_prototype_txout{GetChangePrototypeTxout(fuzzed_data_provider)};
-    coin_params.change_output_size = GetSerializeSize(change_prototype_txout);
+    coin_params.change_output_size = fuzzed_data_provider.ConsumeIntegralInRange(1, MAX_SCRIPT_SIZE);
     coin_params.m_change_fee = effective_fee_rate.GetFee(coin_params.change_output_size);
     coin_params.m_discard_feerate = discard_fee_rate;
     coin_params.change_spend_size = fuzzed_data_provider.ConsumeIntegralInRange<int>(41, 1000);
     const auto change_spend_fee{coin_params.m_discard_feerate.GetFee(coin_params.change_spend_size)};
     coin_params.m_cost_of_change = coin_params.m_change_fee + change_spend_fee;
-    const auto dust{GetDustThreshold(change_prototype_txout, coin_params.m_discard_feerate)};
+    // We only care about the change output size
+    CScript change_out_script;
+    std::fill_n(change_out_script.begin(), coin_params.change_output_size, OP_TRUE);
+    const auto dust{GetDustThreshold(CTxOut{/*nValueIn=*/0, change_out_script}, coin_params.m_discard_feerate)};
     coin_params.min_viable_change = std::max(change_spend_fee + 1, dust);
 
     int next_locktime{0};
```